packages/web/src/pages/artist-coins-explore-page/MobileArtistCoinsExplorePage.tsx

```diff
@@ -0,0 +1,231 @@
+import NavContext, {
+  LeftPreset,
+  RightPreset
+} from 'components/nav/mobile/NavContext'
+
+type CoinRowProps = {
+  coin: Coin
+  onPress: () => void
+}
+
+const CoinRow = ({ coin, onPress }: CoinRowProps) => {
+  const ownerId = HashId.parse(coin.ownerId)
+
+  return (
+    <Flex ph='l' pv='s' alignItems='center' gap='s' onClick={onPress}>
+      <TokenIcon logoURI={coin.logoUri} size='xl' hex />
+
+      <Box flex={1}>
+        <Flex alignItems='center' gap='xs' mb='xs'>
+          <Text variant='title' size='s' strength='weak' ellipses>
+            {coin.name}
+          </Text>
+          <Text variant='body' size='s' color='subdued'>
+            {coin.ticker}
+          </Text>
+        </Flex>
+
+        <UserLink userId={ownerId} size='xs' badgeSize='xs' />
+      </Box>
+    </Flex>
+  )
+}
+
+const NoCoinsContent = () => {
+  return (
+    <Flex
+      direction='column'
+      alignItems='center'
+      justifyContent='center'
+      p='4xl'
+      ph='l'
+      gap='l'
+    >
+      <IconSearch size='2xl' color='default' />
+      <Text variant='heading' size='m' textAlign='center'>
+        {walletMessages.artistCoins.noCoins}
+      </Text>
+      <Text variant='body' size='l' color='subdued' textAlign='center'>
+        {walletMessages.artistCoins.noCoinsDescription}
+      </Text>
+    </Flex>
+  )
+}
+
+const SearchSection = ({
+  searchValue,
+  setSearchValue
+}: {
+  searchValue: string
+  setSearchValue: (value: string) => void
+}) => {
+  return (
+    <Flex borderBottom='default' p='l' backgroundColor='surface1'>
+      <TextInput
+        size={TextInputSize.EXTRA_SMALL}
+        placeholder={walletMessages.artistCoins.searchPlaceholder}
+        value={searchValue}
+        onChange={(e: ChangeEvent<HTMLInputElement>) =>
+          setSearchValue(e.target.value)
+        }
+        label={walletMessages.artistCoins.searchPlaceholder}
+        startIcon={IconSearch}
+      />
+    </Flex>
+  )
+}
+
+export const MobileArtistCoinsExplorePage: React.FC = () => {
+  const dispatch = useDispatch()
+  const location = useLocation()
+  const [searchValue, setSearchValue] = useState('')
+  const [debouncedSearchValue, setDebouncedSearchValue] = useState('')
+  const [sortMethod, setSortMethod] = useState<GetCoinsSortMethodEnum>(
+    GetCoinsSortMethodEnum.MarketCap
+  )
+  const [sortDirection, setSortDirection] = useState<GetCoinsSortDirectionEnum>(
+    GetCoinsSortDirectionEnum.Desc
+  )
+
+  const { data: coins, isPending } = useArtistCoins({
+    sortMethod,
+    sortDirection,
+    query: debouncedSearchValue
+  })
+
+  // Debounce search value to avoid excessive API calls
+  useDebounce(() => setDebouncedSearchValue(searchValue), 300, [searchValue])
+
+  const { setLeft, setRight } = useContext(NavContext)!
+  useEffect(() => {
+    setLeft(LeftPreset.BACK)
+    setRight(RightPreset.KEBAB)
+  }, [setLeft, setRight])
+
+  // Handle route params from sort screen
+  useEffect(() => {
```

Comment:
could we just set the initial values of state from this? or is this expected to change?